src/Backends/REFPROP/REFPROPMixtureBackend.cpp

```diff
@@ -1218,6 +1271,242 @@ phases REFPROPMixtureBackend::GetRPphase()
+	 CoolPropDbl dvdrhoV = -1/POW2(rhoV_mass);
+	 
+	 CoolPropDbl dvL_dp = dvdrhoL*SatL->calc_first_saturation_deriv(iDmass, iP, rhoL_mol, rhoV_mol);
+	 CoolPropDbl dvV_dp = dvdrhoV*SatV->calc_first_saturation_deriv(iDmass, iP, rhoL_mol, rhoV_mol); 
+	 CoolPropDbl dhL_dp = SatL->calc_first_saturation_deriv(iHmass, iP, rhoL_mol, rhoV_mol);
+	 CoolPropDbl dhV_dp = SatV->calc_first_saturation_deriv(iHmass, iP, rhoL_mol, rhoV_mol);
+	 CoolPropDbl dxdp_h = (Q()*dhV_dp + (1 - Q())*dhL_dp)/(hL_mass - hV_mass);
+	 CoolPropDbl dvdp_h = dvL_dp + dxdp_h*(1/rhoV_mol - 1/rhoL_mol) + Q()*(dvV_dp - dvL_dp);
+	 return -POW2(rhomass())*dvdp_h;
+       }
+       else{
+	 throw ValueError("These inputs are not supported to calc_first_two_phase_deriv");
+       }
+}
+
+CoolPropDbl REFPROPMixtureBackend::calc_first_two_phase_deriv_splined(parameters Of, parameters Wrt, parameters Constant, CoolPropDbl x_end) {
+	// Note: If you need all three values (drho_dh__p, drho_dp__h and rho_spline), 
+	// you should calculate drho_dp__h first to avoid duplicate calculations.
+
+	bool drho_dh__p = false;
+	bool drho_dp__h = false;
+	bool rho_spline = false;
+
+	if (Of == iDmolar && Wrt == iHmolar && Constant == iP){
+		drho_dh__p = true;
+		if (_drho_spline_dh__constp) return _drho_spline_dh__constp;
+	}
+	else if (Of == iDmass && Wrt == iHmass && Constant == iP){
+		return first_two_phase_deriv_splined(iDmolar, iHmolar, iP, x_end)*POW2(molar_mass());
+	}
+	else if (Of == iDmolar && Wrt == iP && Constant == iHmolar){
+		drho_dp__h = true;
+		if (_drho_spline_dp__consth) return _drho_spline_dp__consth;
+	}
+	else if (Of == iDmass && Wrt == iP && Constant == iHmass){
+		return first_two_phase_deriv_splined(iDmolar, iP, iHmolar, x_end)*molar_mass();
+	}
+	// Add the special case for the splined density
+	else if (Of == iDmolar && Wrt == iDmolar && Constant == iDmolar){
+		rho_spline = true;
+		if (_rho_spline) return _rho_spline;
+	}
+	else if (Of == iDmass && Wrt == iDmass && Constant == iDmass){
+		return first_two_phase_deriv_splined(iDmolar, iDmolar, iDmolar, x_end)*molar_mass();
+	}
+	else{
+		throw ValueError("These inputs are not supported to calc_first_two_phase_deriv");
+	}
+
+       if (_RPcheckTwophase() == false)
+	 throw ValueError("Not two phase fluid: saturation derivative is not possible");
+    
+       shared_ptr<REFPROPMixtureBackend> SatL(new REFPROPMixtureBackend(this->calc_fluid_names())), SatV(new REFPROPMixtureBackend(this->calc_fluid_names()));
+       SatL->set_mole_fractions(this->get_mole_fractions());
+       SatV->set_mole_fractions(this->get_mole_fractions());
+       SatL->update(QT_INPUTS, 0.0, _T); // <--- This looks never effecient,
+       SatV->update(QT_INPUTS, 1.0, _T); //      but just mimics the saturate states of HEOS..
+
+       shared_ptr<REFPROPMixtureBackend> Liq(new REFPROPMixtureBackend(this->calc_fluid_names())), End(new REFPROPMixtureBackend(this->calc_fluid_names()));
+    
+    Liq->specify_phase(iphase_liquid);
```

Comment:
```suggestion
    Liq->set_mole_fractions(this->get_mole_fractions());
    Liq->specify_phase(iphase_liquid);
```
Missing mole_fractions